src/nvim/search.c

```diff
@@ -4347,121 +4347,288 @@ int linewhite(linenr_T lnum)
 }
 
 // Add the search count "[3/19]" to "msgbuf".
-// When "recompute" is true Always recompute the numbers.
-static void search_stat(int dirc, pos_T *pos,
-                        bool show_top_bot_msg, char_u *msgbuf, bool recompute)
+// See update_search_stat() for other arguments.
+static void cmdline_search_stat(int dirc, pos_T *pos, pos_T *cursor_pos,
+                                int show_top_bot_msg, char_u  *msgbuf,
+                                int recompute, int maxcount, long timeout)
 {
-    int       save_ws = p_ws;
-    int       wraparound = false;
-    pos_T     p = (*pos);
-    static  pos_T   lastpos = { 0, 0, 0 };
+    searchstat_T stat;
+
+    update_search_stat(dirc, pos, cursor_pos, &stat, recompute, maxcount,
+                       timeout);
+    if (stat.cur > 0) {
+      char  t[SEARCH_STAT_BUF_LEN];
+      size_t  len;
+
+      if (curwin->w_p_rl && *curwin->w_p_rlc == 's') {
+        if (stat.incomplete == 1) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[?/??]");
+        } else if (stat.cnt > maxcount && stat.cur > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[>%d/>%d]",
+                       maxcount, maxcount);
+        } else if (stat.cnt > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[>%d/%d]",
+                       maxcount, stat.cur);
+        } else {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[%d/%d]",
+                       stat.cnt, stat.cur);
+        }
+      } else {
+        if (stat.incomplete == 1) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[?/??]");
+        } else if (stat.cnt > maxcount && stat.cur > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[>%d/>%d]",
+                       maxcount, maxcount);
+        } else if (stat.cnt > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[%d/>%d]",
+                       stat.cur, maxcount);
+        } else {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[%d/%d]",
+                       stat.cur, stat.cnt);
+        }
+      }
+
+      len = STRLEN(t);
```

Comment:
`size_t len = strlen(t);` because `t` is `char` type.

Reply:
But len is alrwady declared as size_t at line 4361

Reply:
Done